src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -0,0 +1,534 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#include <array>
+#include <cstring>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <filesystem>
+
+#ifdef _WIN32
+#include "getopt.h"
+#include <shlwapi.h>
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+// Global mutexes to ensure thread safety when accessing shared resources
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+std::mutex env_mutex;
+std::mutex getopt_mutex;
```

Comment:
Nah.  These globals are not cool.   What shared resources are these protecting?   Only within this file, then fine, but put them in unnamed namespace.  If these are real cross compilation unit globals, then we need to rethink a better way.

I do think all your project code that spans multiple compilation units should be within a namespace.  Since this is under XRT I think the namespace should be
```
namespace xrt_core::xbtracer {
}
```

Reply:
Done, please review